Add jnp.matrix_transpose() and jax.Array.mT

jakevdp opened the pull request:
This is an API proposed by the Python Array API Standard (https://data-apis.org/array-api/2022.12/). It's lightweight enough that there's hardly any downside to supporting it in JAX.

Related to #16099

hawkinsp commented on jax/_src/numpy/lax_numpy.py:
```diff
@@ -538,6 +538,28 @@ def transpose(a: ArrayLike, axes: Optional[Sequence[int]] = None) -> Array:
 
+@util._wraps(getattr(np, 'matrix_transpose', None))
+def matrix_transpose(x: ArrayLike, /) -> Array:
+  """Transpose the last two dimensions of x.
```
Nit: "Transposes"

jakevdp replied:
Done

hawkinsp reviewed (approved):
Consider replacing the `_T` helper used in the following two places:
* https://github.com/google/jax/blob/e464dc870054c86bb6fee4ecca2ebf066f32ade5/jax/_src/numpy/linalg.py#L36
* https://github.com/google/jax/blob/e464dc870054c86bb6fee4ecca2ebf066f32ade5/jax/_src/lax/linalg.py#L381

?

jakevdp committed:
Add jnp.matrix_transpose() and jax.Array.mT

This is an API proposed by the Python Array API Standard (https://data-apis.org/array-api/2022.12/). It's lightweight enough that there's hardly any downside to supporting it in JAX.

jakevdp committed:
Use new matrix_transpose in linalg code

jakevdp commented:
Thanks - I added a second commit using `mT` in the linalg code.
